src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/AsylumCaseNotificationHandler.java

```diff
@@ -52,13 +53,16 @@ public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCa
         final String caseId = asylumCase.read(APPEAL_REFERENCE_NUMBER, String.class)
             .orElseThrow(() -> new IllegalStateException("Case ID for the appeal is not present"));
 
-        MessageHeader messageHeader = homeOfficeInstructService.sendNotification(homeOfficeReferenceNumber,caseId);
+        final String correlationId = HomeOfficeRequestHelper.generateUuid();
+        HomeOfficeInstructResponse instructResponse = homeOfficeInstructService
+            .sendNotification(homeOfficeReferenceNumber, caseId, correlationId);
+
         PreSubmitCallbackResponse<AsylumCase> response = new PreSubmitCallbackResponse<>(asylumCase);
-        if (messageHeader == null) {
-            asylumCase.write(HOME_OFFICE_INSTRUCT_STATUS,false);
+        if (instructResponse == null || instructResponse.getMessageHeader() == null) {
```

Comment:
Can you not do this kind of stuff in the Feign decoder so that when you get to this point there is no checking to do?  If there's an error it should either throw an exception or be handled.

Also I'm not sure you know if this is an Internal Server Error at this point?

Reply:
Also having just looked at the FeignDecoder I'm not sure this code will ever have a null response here.  I think we can leave it as it is for now and when Jag does the integration tests we can test it out.

Reply:
For now, I will leave it as it is